src/XrmMockupShared/Utility.cs

```diff
@@ -546,25 +546,46 @@ public static Entity MakeProxyTypeEntity(Entity entity, Type ty)
             return (Entity)method.Invoke(entity, null);
         }
 
-        public static bool MatchesCriteria(DbRow row, FilterExpression criteria)
+        public static bool MatchesCriteria(Entity row, FilterExpression criteria)
```

Comment:
It seems counter intuitive to move away from using DbRow. By using the row directly we avoid converting the entity to a row and back throughout XrmMockup. Therefore we should strive towards only using DbRow inside XrmMockup.